tests/test_shell.py

```diff
@@ -188,6 +250,68 @@ def test_find(self):
                                  ['{"a": ' + str(i) + '}']).strip() == \
                     list(project.find_job_ids(doc_filter={'a': i}))[0]
 
+    def test_diff(self):
+        self.call('python -m signac init ProjectA'.split())
+        project = signac.Project()
+        job_a = project.open_job({"a": 0, "b": 1})
+        job_a.init()
+        job_b = project.open_job({"a": 0, "b": 0})
+        job_b.init()
+        out = self.call('python -m signac diff {} {}'.format(job_a.id, job_b.id).split())
+        expected = [str(job_a.id), "{'b': 1}", str(job_b.id), "{'b': 0}"]
+        assert out.strip().split(os.linesep) == expected
+
+    def test_clone(self):
+        self.call('python -m signac init ProjectA'.split())
+        project_a = signac.Project()
+        project_b = signac.init_project('ProjectB', os.path.join(self.tmpdir.name, 'b'))
+        job = project_a.open_job({'a': 0})
+        job.init()
+        assert len(project_a) == 1
+        assert len(project_b) == 0
+        with pytest.raises(ExitCodeError):
+            self.call("python -m signac clone {} 9bfd29df07674bc5"
```

Comment:
Can we add a comment here about the meaning of that hard-coded id? Is it expected to already exist?